File: LeetCode-Programs/Best Time to Buy and Sell Stock.py
class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        maxp=0
        temp=prices[0]                                 
        for i in range(len(prices)): 
            if temp>prices[i]:
                temp=prices[i]
                
            if (prices[i]-temp) > maxp:
                maxp=prices[i]-temp
        return maxp;

File: LeetCode-Programs/Best Time to Buy and Sell Stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
    int temp,maxp=0;
    temp=prices[0];                                  
    for(int i=0;i<prices.size();i++) 
    {
        if(temp>prices[i])                                  
            
            temp=prices[i];                                   
        
       if( (prices[i]-temp) > maxp)                          
           maxp=prices[i]-temp;                              
    }
    return maxp;
}
    
};
